packages/components/src/icon-button/index.js

```diff
@@ -18,9 +18,19 @@ import Dashicon from '../dashicon';
 
 // This is intentionally a Component class, not a function component because it
```

Comment:
From the comment here, and the introduction of `forwardRef`, I wonder if it's still necessary for this to be a class. Having it be a function instead simplifies greatly the `ref` forwarding, since it's provided as the second argument to the function (see [`button/index.js`](https://github.com/WordPress/gutenberg/blob/c3cacd3d88014c913478d76e8bc24c849638f848/packages/components/src/button/index.js#L11)).

Let me test to see if that'll work...